Sources/Core/State/PlayingState.swift

```diff
@@ -140,6 +146,13 @@ final class PlayingState: PlayerState {
             }
         }
     }
+
+    private func sendError(for playerItem: AVPlayerItem?, with delegate: PlayerContextDelegate?) {
+        let errorLog = playerItem?.errorLog()
+        let accessLog = playerItem?.accessLog()
+        let playerItemErrorInfo = PlayerItemErrorInfo(errorLog: errorLog, accessLog: accessLog)
+        delegate?.playerContext(itemErrorInfo: playerItemErrorInfo)
```

Comment:
We have here to print log messages in the error domain.
We can also expose something to the user, I'm ok with this but not String either AVPlayer references.
What do you think about an public enum ?

Reply:
It's not just print log messages, in our case we have to do custom actions with the errors in the client side. My initial choice was to add only the errorCode and the errorMessage (I thought it was enough) but you didn't liked this solution. I changed to add all the error object (even if I don't like a lot like you). What properties from the errors do you want to expose to the client? Do you think that an enum is more flexible than a struct? When you need to add another cases in the future is always a major update for the client, otherwise with a struct you can add new properties and with adding default inits you can create a minor change for the client. In the other hand, changing an enum will be always a major update for the client. But it's just an implementation detail, I change to an enum no problem :)
If I understand correctly, you want an enum with two cases for errorLog and accessLog? Can you specify which associated values do you want for the cases? The delegate method will need to be called twice in that case (for the two cases), it's okay for you?

Reply:
ok, to totally understand the situation, I need more explanation:

1. you call the delegate in 2 situations when playing failed -> so the state changed. Client side really need additional logs form AVPlayer Item to do something different ? 

2. like I said, I'm agree to print logs and expose something in client side.

3. public enum you're right is not so "SOLID". so we may have think about a Struct but not a specific AVPlayerItemError's Struct .  Something more global, what do you think of this:

`public struct ModernAVPlayerLog {
    let date: Date
    let domain: ModernAVPlayerLoggerDomain
    let statusCode: Int?
    let content: String?
}
`
So we can merge AVPlayerItem Errors & in the future other debug message..
I know about the enum in the struct... but I think it's relevant than a String.